src/rpc/server/linux_io_uring.zig

```diff
@@ -0,0 +1,835 @@
+                            .which = .to_pipe,
+                        } };
+                        const sfb = &entry_data.state.send_file_body;
+                        try sfb.prepSpliceFileToPipe(entry, &liou.io_uring);
+                        return .ok;
+                    },
+                },
+            }
+        },
+
+        .send_file_body => |*sfb| switch (sfb.which) {
+            .to_pipe => {
+                switch (try connection.handleSpliceResult(cqe.err())) {
+                    .success => {},
+                    .again => std.debug.panic(EAGAIN_PANIC_MSG, .{}),
+                    .bad_file_descriptors,
+                    .bad_fd_offset,
+                    .invalid_splice,
+                    => return .splice_error,
+                }
+                sfb.spliced_to_pipe += @intCast(cqe.res);
+
+                sfb.which = .to_socket;
+                try sfb.prepSplicePipeToSocket(entry, &liou.io_uring);
+
+                return .ok;
+            },
+            .to_socket => {
+                switch (try connection.handleSpliceResult(cqe.err())) {
+                    .success => {},
+                    .again => std.debug.panic(EAGAIN_PANIC_MSG, .{}),
+                    .bad_file_descriptors,
+                    .bad_fd_offset,
+                    .invalid_splice,
+                    => return .splice_error,
+                }
+                sfb.spliced_to_socket += @intCast(cqe.res);
+
+                if (sfb.spliced_to_socket < sfb.sfd.file_size) {
+                    sfb.which = .to_pipe;
+                    try sfb.prepSpliceFileToPipe(entry, &liou.io_uring);
+                } else {
+                    std.debug.assert(sfb.spliced_to_socket == sfb.spliced_to_pipe);
+                    entry.deinit(server_ctx.allocator);
+                    server_ctx.wait_group.finish();
+                }
+                return .ok;
+            },
+        },
+
+        .send_no_body => |*snb| {
+            switch (try connection.handleSendResult(cqe.err())) {
+                .success => {},
+                .again => std.debug.panic(EAGAIN_PANIC_MSG, .{}),
+                .intr => return .signal_interrupted_operation,
+                .conn_reset => return .misc_connection_failure,
+            }
+            const sent_len: usize = @intCast(cqe.res);
+            snb.end_index += sent_len;
+
+            if (snb.end_index < snb.head.len) {
+                try snb.prepSend(entry, &liou.io_uring);
+                return .ok;
+            } else std.debug.assert(snb.end_index == snb.head.len);
+
+            entry.deinit(server_ctx.allocator);
+            server_ctx.wait_group.finish();
+            return .ok;
+        },
+    }
+
+    comptime unreachable;
+}
+
+const HandleRecvBodyError =
+    IouGetSqeError ||
+    std.fs.Dir.StatFileError ||
+    std.fs.File.OpenError ||
+    std.fs.File.GetSeekPosError ||
+    std.posix.PipeError;
+fn handleRecvBody(
```

Comment:
I think it's easier to read with a blank line. I see you've done this in other places too. Curious what others think.
```suggestion
    std.posix.PipeError;

fn handleRecvBody(
```

Reply:
i pref the blank lines too

Reply:
Small oversight, changed in 1242b40725d54ee352f8f47cb860851b790bfe82